src/libcore/cmp.rs

```diff
@@ -1126,7 +1127,16 @@ mod impls {
     impl Ord for bool {
         #[inline]
         fn cmp(&self, other: &bool) -> Ordering {
-            (*self as u8).cmp(&(*other as u8))
+            // Casting to i8's and converting the difference to an Ordering generates
+            // more optimal assembly.
+            // See <https://github.com/rust-lang/rust/issues/66780> for more info.
+            match (*self as i8) - (*other as i8) {
+                -1 => Less,
+                0 => Equal,
+                1 => Greater,
+                // SAFETY: Unreachable code
```

Comment:
This doesn't seem like it's really doing anything. The unsafe code is calling `unreachable_unchecked` - of course it's unreachable code!

Reply:
I had to document the unsafe block using the safety comment for the PR checks to pass. I can add a better comment but that's the reason for that comment.

Reply:
If we're going to require documentation for unsafe blocks, it may as well be useful documentation. e.g. "SAFETY: bool as i8 returns 0 or 1, so the subtraction can't end up with anything else."